New Feature/actions file menu

Abijeet opened the pull request:
Towards: https://github.com/nextcloud/server/issues/7647

As described, have created a new drop-down menu shown when multiple files are selected. I also modified the trashbin plugin (Deleted files), I've put the restore and delete items under the actions menu.

Here are a few screenshots,

## Laptop

![screenshot from 2018-06-02 21-02-40](https://user-images.githubusercontent.com/1685517/40878028-540fb8ae-66a8-11e8-92fe-e2d464670a4e.png)

### Deleted files

![screenshot from 2018-06-02 21-29-33](https://user-images.githubusercontent.com/1685517/40878266-1701582e-66ac-11e8-9d2e-de7f8284768b.png)

## Mobile

### Landscape
![localhost_nextcloud_index php_apps_files__dir _ iphone 6_7_8 plus](https://user-images.githubusercontent.com/1685517/40878020-2da19570-66a8-11e8-9a73-ceeecec7abcb.png)

### Potrait
![localhost_nextcloud_index php_apps_files__dir _ iphone 6_7_8 plus 1](https://user-images.githubusercontent.com/1685517/40878032-65f70bda-66a8-11e8-9785-c5cb82b0dab8.png)

Abijeet committed:
Adds an action dropdown that is shown on file selection.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet committed:
Added a new action menu in files and trash list.

Uses the new file-multi-select-menu component.

Towards #7647

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet committed:
Fixes the position of the navigation toggle menu icon in mobile.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet commented on core/css/mobile.scss:
```diff
@@ -67,7 +67,7 @@
 	position: fixed;
 	display: inline-block !important;
-	top: 45px;
+	top: 50px;
```
This change was made to fix an issue I noticed with the navigation toggle on mobile resolutions,

## Without fix
![localhost_nextcloud_index php_apps_files__dir _ view trashbin pixel 2 xl](https://user-images.githubusercontent.com/1685517/40878326-fc2548ca-66ac-11e8-8205-3ae6e087c35e.png)

## With fix

![localhost_nextcloud_index php_apps_files__dir _ view trashbin pixel 2 xl 1](https://user-images.githubusercontent.com/1685517/40878328-012bc614-66ad-11e8-90bc-fcf1b99b39e5.png)

jancborchardt replied:
Thanks, good catch! That’s because we increased the height of the header recently. :)

jancborchardt commented:
This looks great! :) Please also review @nextcloud/designers @nextcloud/javascript

skjnldsv commented:
@jancborchardt not sure it's ready to review since @Abijeet left the WIP in the title. :)

jancborchardt reviewed (commented):
-

jancborchardt commented:
@skjnldsv hmm, can’t seem to withdraw the review? :\

skjnldsv commented:
@jancborchardt you need to request changes indeed. You cannot 'remove' the review :)

Abijeet commented:
I'll work on fixing the test cases.

Abijeet committed:
Fixed failing test cases for the new actions menu.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet committed:
Fixes issues found during running of test cases.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

MorrisJobke commented:
Please rebase on the latest master to get the fixes for the mysql failures ;)

Abijeet commented:
> Please rebase on the latest master to get the fixes for the mysql failures ;)

@MorrisJobke  - Thanks. I've done that and that seems to have fixed the MySQL failures. Not sure about the other failing issue, I think it's not related to my changes

rullzer commented:
@Abijeet since you changed the title and the tests pass I assume this is good to review

rullzer commented:
@juliushaertl @skjnldsv @danxuliu work your review magic!

juliushaertl commented:
@Abijeet Really nice improvement :+1: 

Can you have a look at the positioning of the text when selecting an element, it seems to be some pixels above the text/checkbox when no element is selected.

![peek 2018-06-11 09-36](https://user-images.githubusercontent.com/3404133/41218375-282a0758-6d5b-11e8-8064-a954a8c412f8.gif)

Abijeet commented:
Hey @juliushaertl,
> Can you have a look at the positioning of the text when selecting an element, it seems to be some pixels above the text/checkbox when no element is selected.

I'm sorry I noticed this and it subsequently skipped my mind. I'll get round to fixing it soon.

Abijeet commented:
I've fixed the issues with vertical moving. I'm also noticing that the name column expands slightly when a row is selected. Its because the Size column on the top which contains the size of the file (that appears when a file is selected) is wider when compared to the column heading "Size". 

We also have a hidden (not `display: none`) sort icon next to the file size which if you notice is causing the total size not to be aligned with the individual file size options. I assume we used `visibility: hidden` instead of `display: none` in order to not disturb the size of the columns on selection.

![sideway-movings](https://user-images.githubusercontent.com/1685517/41267704-24cfaba2-6e1a-11e8-961b-566f51f4249e.gif)

Abijeet committed:
Fixes UI issues with the design,

1. Removes an unnecessary width
2. Content of the table header moving down on selection. Happening due to increase in height of header.
3. Invalid font-size for the actions menu.
4. Vertical position of the action menu

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet commented:
Hey @juliushaertl 

This is ready for another review. I've made some other UI tweaks since you last reviewed it. 

I also spent some time on the issue below but could not fix it,

![sideway-movings](https://user-images.githubusercontent.com/1685517/41267704-24cfaba2-6e1a-11e8-961b-566f51f4249e.gif)

I'm fairly confident that this was not caused due to my changes so I think we can go ahead and merge this if you guys think its ready.

Let me know if you want me to create another issue to handle the bug in the GIF.

MorrisJobke committed:
Fix alignment of more icon

Signed-off-by: Morris Jobke <hey@morrisjobke.de>

MorrisJobke commented:
I fixed the alignment of the more icon:

Before:

![bildschirmfoto 2018-06-15 um 17 08 57](https://user-images.githubusercontent.com/245432/41475597-82fc5282-70bf-11e8-93fd-b5f957a96398.png)

After:

![bildschirmfoto 2018-06-15 um 17 08 53](https://user-images.githubusercontent.com/245432/41475596-82e2823a-70bf-11e8-8337-f031ababd586.png)

There is one little glitch remaining:

* [ ] when there is no file selected the actions menu is visible behind the file list
![bildschirmfoto 2018-06-15 um 17 13 16](https://user-images.githubusercontent.com/245432/41475598-8316de0e-70bf-11e8-98c1-f22de4ce99fe.png)

* [ ] when start doing a move and then abort the dialog (instead of click move and copy) then the loading circle is still present in the menu

Beside that it looks really good 👍

Abijeet committed:
Fixes issue with action being displayed incorrectly.

Also fixes the issue with the loading icon on copy & move.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet committed:
Fixes the restore loading icon appearing continuously.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet commented:
Hey @MorrisJobke,

Thanks for checking this out.

>  when there is no file selected the actions menu is visible behind the file list

I should have noticed this. It was caused due to my latest changes. Fixed it.

>  when start doing a move and then abort the dialog (instead of click move and copy) then the loading circle is still present in the menu

This is also fixed. A similar issue was present on the Deleted files restore icon as well, which I have fixed too.

I've added some more assertions to avoid some of the issues from happening.

Abijeet committed:
Adds a test case for the loading symbol in deleted files.

Signed-off-by: Abijeet <abijeetpatro@gmail.com>

Abijeet commented:
Re-basing with the master branch does not seem to have fixed the build failures.

Build failing - **TESTS=vue-builds**

Error,
```
./build/vue-builds.sh ./settings/js/main.js
The build file ./settings/js/main.js does not exists 
```

skjnldsv commented:
> This branch is 10 commits ahead, 54 commits behind master.
I guess the rebase failed :)
